File: SDKs/beezup.com/2.0/typescript/src/sdk/models/shared/importationcustomcolumn.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ColumnConfiguration } from "./columnconfiguration";
import { ImportationCustomColumnLinks } from "./importationcustomcolumnlinks";



// ImportationCustomColumn
/** 
 * The custom column configuration
**/
export class ImportationCustomColumn extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=configuration" })
  configuration: ColumnConfiguration;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=links" })
  links: ImportationCustomColumnLinks;

  @SpeakeasyMetadata({ data: "json, name=userColumName" })
  userColumName: string;
}

File: SDKs/beezup.com/2.0/python/src/sdk/models/shared/importationcustomcolumn.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import columnconfiguration as shared_columnconfiguration
from ..shared import importationcustomcolumnlinks as shared_importationcustomcolumnlinks


@dataclass_json
@dataclasses.dataclass
class ImportationCustomColumn:
    r"""ImportationCustomColumn
    The custom column configuration
    """
    
    configuration: shared_columnconfiguration.ColumnConfiguration = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('configuration') }})
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    links: shared_importationcustomcolumnlinks.ImportationCustomColumnLinks = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('links') }})
    user_colum_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('userColumName') }})
    
